src/AppInstallerCommonCore/Registry.cpp

```diff
@@ -439,6 +439,79 @@ namespace AppInstaller::Registry
         return result;
     }
 
+    Key Key::CreateKeyAndOpen(HKEY key, std::string_view subKey, DWORD options, REGSAM access)
+    {
+        return CreateKeyAndOpen(key, Utility::ConvertToUTF16(subKey), options, access);
+    }
+
+    Key Key::CreateKeyAndOpen(HKEY key, const std::wstring& subKey, DWORD options, REGSAM access)
+    {
+        Key result;
+        result.Create(key, subKey, options, access);
+        return result;
+    }
+
+    bool Key::DeleteKey(HKEY key, std::string_view subKey, DWORD samDesired)
+    {
+        return DeleteKey(key, Utility::ConvertToUTF16(subKey), samDesired);
+    }
+
+    bool Key::DeleteKey(HKEY key, const std::wstring& subKey, DWORD samDesired)
+    {
+        LSTATUS status = RegDeleteKeyExW(key, subKey.c_str(), samDesired, 0);
+        if (status == ERROR_SUCCESS)
+        {
+            AICLI_LOG(Core, Verbose, << "Subkey '" << Utility::ConvertToUTF8(subKey) << "' was deleted successfully.");
+            return true;
+        }
+
+        THROW_IF_WIN32_ERROR(status);
+        return false;
+    }
+
+    bool Key::Create(HKEY key, const std::wstring& subKey, DWORD options, REGSAM access)
+    {
+        m_access = access;
+        LPDWORD lpdwDisposition = {};
+        LSTATUS status = RegCreateKeyExW(key, subKey.c_str(), 0, nullptr, options, access, NULL, &m_key, lpdwDisposition);
+
+        if (lpdwDisposition == (LPDWORD)REG_CREATED_NEW_KEY)
+        {
+            AICLI_LOG(Core, Verbose, << "Subkey '" << Utility::ConvertToUTF8(subKey) << "' was created.");
+        }
+        else if (lpdwDisposition == (LPDWORD)REG_OPENED_EXISTING_KEY)
+        {
+            AICLI_LOG(Core, Verbose, << "Subkey '" << Utility::ConvertToUTF8(subKey) << "' already existed and was opened.");
+        }
+
+        THROW_IF_WIN32_ERROR(status);
+        return true;
+    }
+
+    bool Key::TrySetRegistryValue(wil::shared_hkey key, const std::wstring& name, const std::wstring& value, DWORD& dwType)
+    {
+        LSTATUS status;
+        if (name.empty())
+        {
+            status = RegSetValueExW(key.get(), NULL, 0, dwType, (LPBYTE)value.c_str(), (DWORD)(value.length() + 1) * sizeof(wchar_t));
+        }
+        else
+        {
+            status = RegSetValueExW(key.get(), name.c_str(), 0, dwType, (LPBYTE)value.c_str(), (DWORD)(value.length() + 1) * sizeof(wchar_t));
+        }
+
+        if (status == ERROR_SUCCESS)
+        {
+            AICLI_LOG(Core, Verbose, << "Set registry name '" << Utility::ConvertToUTF8(name) << "' with the value '" << Utility::ConvertToUTF8(value));
+            return true;
+        }
+        else
+        {
+            AICLI_LOG(Core, Verbose, << "Failed to set name value '" << Utility::ConvertToUTF8(name));

```

Comment:
Throw on error, don't use bool returns.

Which is to say, yes a `Try...` function should use a return value, but there is no code calling this that is actually using that value.  So this function should not be a "try" and should throw on error.

Reply:
Changed to throwing on error instead of returning a bool